contrib/fparser/fparser_ad.cc

```diff
@@ -22,6 +22,16 @@ using namespace FPoptimizer_CodeTree;
 
 #include "lib/sha1.h"
 
+// There are several case statements in this file where we
+// intentionally want to fall through, so let's not get warned about
+// each one.
+#ifdef __GNUC__
+#if (__GNUC__ > 6)
+#  pragma GCC diagnostic ignored "-Wimplicit-fallthrough"
+#endif
+#endif
+
+
```

Comment:
Why not just include the new header here (and in the other half dozen .cc files)?  Trying to keep the other two warnings active?

Reply:
Yeah, principle of least change I guess. Since we aren't currently getting those other warnings from this file, I'd rather not disable them yet.